### PR TITLE
Add-on functionality for conservation list handling

### PR DESCRIPTION
**Pull request by reupost**
These changes allow for the following:
* Recording simple species membership of a species list, rather than a specific KVP on the list
* Appending values to multivalue SOLR fields
* Initial deletion of all values for a particular field in SOLR, to allow one to remove the statuses of species that have been removed from a list

All functionality is opt-in via the conservation-lists.json configuration file.

Additional configuration options are:

"deleteFirst": "[comma-delimited list of fields to set to null before processing]"
- fields to set to null across entire BIE SOLR instance

[lists] "sourceField: "*" 
- record list membership (as list name) rather than specific source field KVP

An example configuration file is as follows:

{
  "defaultSourceField": "status",
  "deleteFirst": "listMembership,riskAssessment,managementPlans",
  "lists": [
    {
      "uid": "dr1818",
      "sourceField": "*",
      "field": "listMembership",
      "action": "add",
      "term": "GB NNSS ALERT SPECIES",
      "label": "INNS"
    },
    {
      "uid": "dr1825",
      "sourceField": "*",
      "field": "listMembership",
      "action": "add",
      "term": "WCA 1981 SCHEDULE 9",
      "label": "INNS"
    },
    {
      "uid": "dr1831",
      "sourceField": "*",
      "field": "listMembership",
      "action": "add",
      "term": "INNS - WALES PRIORITY SPECIES FOR ACTION",
      "label": "INNS"
    },
    {
      "uid": "dr1826",
      "sourceField": "innsRiskAssessment",
      "field": "riskAssessment",
      "action": "add",
      "term": "INNS risk assessment: WELSH MARINE MONITORING AND SURVEILLANCE LIST",
      "label": "INNS"
    },
    {
      "uid": "dr1831",
      "sourceField": "Priority",
      "field": "managementPlans",
      "action": "add",
      "term": "INNS management plans: INNS - WALES PRIORITY SPECIES FOR ACTION",
      "label": "INNS"
    }
  ]
}

### CODE REVIEW

**Comment by charvolant** [comment]
Conflicts first:

The gradle.build file has changing: true on non-snapshot dependencies. We've been asked by DAWR (Martin, really) to ensure that release versions of the bie are nailed down to have non-changing, release dependencies to avoid build chaos.

The approach to the highlighting is better than just a string.

File: grails-app/services/au/org/ala/bie/SearchService.groovy [hunk 1]
```diff
@@ -870,7 +867,6 @@ class SearchService {
                             namePublishedInYear: synonym.namePublishedInYear,
                             namePublishedInID: synonym.namePublishedInID,
                             nameAuthority: synonym.datasetName ?: datasetName ?: grailsApplication.config.synonymSourceAttribution,
-                            taxonRemarks: synonym.taxonRemarks,
```
Comment (charvolant) [inline grails-app/services/au/org/ala/bie/SearchService.groovy]: These need to be included

File: grails-app/services/au/org/ala/bie/SearchService.groovy [hunk 2]
```diff
@@ -847,7 +845,6 @@ class SearchService {
                         namePublishedIn: taxon.namePublishedIn,
                         namePublishedInYear: taxon.namePublishedInYear,
                         namePublishedInID: taxon.namePublishedInID,
-                        taxonRemarks: taxon.taxonRemarks,
```
Comment (charvolant) [inline grails-app/services/au/org/ala/bie/SearchService.groovy]: These need to be included

File: grails-app/services/au/org/ala/bie/SearchService.groovy [hunk 3]
```diff
@@ -883,16 +879,6 @@ class SearchService {
                             status: commonName.status,
                             priority: commonName.priority,
                             language: commonName.language ?: grailsApplication.config.commonNameDefaultLanguage,
-                            temporal: commonName.temporal,
```
Comment (charvolant) [inline grails-app/services/au/org/ala/bie/SearchService.groovy]: Part of the IEK work. Common names can now have details about what the name applies to,

**Comment by djtfmartin** [comment]
thanks for looking at this @charvolant 

@reupost are you able to update/merge your fork ?

**Comment by reupost** [comment]
Hi @charvolant, thanks for going through this. Happy to make changes to fix it up, but a bit confused by the way github is displaying things (I haven't done this before, sorry!). I'm not sure exactly what I need to update. It looks like you've already made some of the corrections, and/or have you merged the latest ALA code with the fork? I don't know anything about the IEK stuff - I didn't change any of that as far as I know. Is it new since I made the pull request 3 months ago?

**Comment by charvolant** [comment]
Hi @reupost It looks like your pull request is removing a number of fields, particularly in the common name area where terms like temporal, lifeStage, etc. Seem to have gone missing. However, I've just checked and they don't seem to be missing any more, so I'm not sure what what's going on.

**Comment by adam-collins** [comment]
An update
1. Multiple config `bq`s are supported.
2. The config for `hl` has changed. 
3. The plan is to remove this configuration and defer to lists. This is consistent with pipelines. https://github.com/AtlasOfLivingAustralia/bie-index/issues/130
4. I do wonder if the aggregation of the conservationStatus map would be more easily done in the client application. 
5. For searching of conservationStatus the use of a multivalue SOLR field might be more appropriate.

**Pull request closed without merging**